### DISCUSSION TITLE
SCIP Relative MIP Relative Min Gap: Different Objective Values per Run

### DISCUSSION THREAD

**Discussion by Jiw0on** [google/or-tools#3417] [Linear Solver questions]
Hey all,

Currently I'm running a project where I utilize the relative min gap parameter to find the best possible solution under specified time limit and gap of 10%. 

gap = 0.1
solverParams = pywraplp.MPSolverParameters()
solverParams.SetDoubleParam(solverParams.RELATIVE_MIP_GAP, gap)

When I run various instances of the LP, I get drastically different objective values ranging anywhere from 9100 to 300000, where 300000 is what I ideally like to be around as it maximizes my objective function. My constraints do not change at any point . I understand the concepts associated with the  relative gap defined by SCIP which is defined as:

relative gap = |primal - dual|/MIN(|dual|,|primal|),

but my underlying question is if the dual constructed by the program is cherry picking constraints that place it at a local maxima which is the reason why the objective value sometimes hangs around 9000 depending on the run. I've experimented with different time limits and different gap values, but the results seem to vary among all the runs. Any sort of insights would be very much appreciated.

Thank you!

**Comment by lperron** [comment]
why don't you ask the scip team ? Are you hitting the time limit ?

**Reply by Jiw0on** [reply]
That's fair. I'll also shoot them a message. And yes, I've been hitting the time limit. I've experimented at 5, 10, 30 , 45 and 60 minutes, and all yield variable objective values.

**Reply by lperron** [reply]
There are different LPs ? In that case, this is expected.
I thought you were running the same problem with vastly different results.

**Reply by Jiw0on** [reply]
No this is the same exact LP. I'm running through the same objective function with the same constraints and decision variables but with different parameters for time limits and relative MIP gap, and still getting vastly different results.

**Reply by lperron** [reply]
it is quite possible that scip allocates parts of the time limit to different phases, allowing vastly different results.
Something like 20% of the time on the first solution heuristic. 10% on probing...

Thus a different time limit means very different results.
